mutalyzer/variant.py

```diff
@@ -0,0 +1,270 @@
+"""
+"""
+
+from __future__ import unicode_literals
+
+from Bio.SeqUtils import seq3
+
+from extractor import extractor
+
+from mutalyzer import models
+
+weights = {
+    "subst": extractor.WEIGHT_SUBSTITUTION,
+    "del": extractor.WEIGHT_DELETION,
+    "ins": extractor.WEIGHT_INSERTION,
+    "dup": extractor.WEIGHT_INSERTION,
+    "inv": extractor.WEIGHT_INVERSION,
+    "delins": extractor.WEIGHT_DELETION_INSERTION
+}
+
+class HGVSList(list):
```

Comment:
I read the chapter from Python the hard way.

The reason I chose to subclass `list` is because we need iteration, indexing and slicing. Furthermore, if we want to combine allele descriptions, we probably need a function like pop() as well. It seems to me that this reduces the amount of code.

The behaviour you describe is indeed troublesome, I'll give composition a try.